presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -50,11 +50,7 @@ public IterativeOptimizer(StatsRecorder stats, List<PlanOptimizer> legacyRules,
 
     public IterativeOptimizer(StatsRecorder stats, Set<Rule> rules)
     {
-        this.legacyRules = ImmutableList.of();
-        this.rules = ImmutableSet.copyOf(rules);
-        this.stats = stats;
-
-        stats.registerAll(rules);
+        this(stats, ImmutableList.of(), rules);
```

Comment:
this constructor should be at the top, as it is using the other one. That way code usage order would match the order in which code is read.